pkg/driver/kubernetes_driver.go

```diff
@@ -0,0 +1,366 @@
+			Name: k8sFileSecretVolume,
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: secret.ObjectMeta.Name,
+				},
+			},
+		})
+		container.VolumeMounts = mounts
+	}
+
+	job.Spec.Template.Spec.Containers = []v1.Container{container}
+	job, err := k.jobs.Create(job)
+	if err != nil {
+		return err
+	}
+	if !k.SkipCleanup {
+		defer k.deleteJob(job.ObjectMeta.Name)
+	}
+
+	// Return early for unit testing purposes (the fake k8s client implementation just
+	// hangs during watch because no events are ever created on the Job)
+	if k.skipJobStatusCheck {
+		return nil
+	}
+
+	selector := metav1.ListOptions{
+		LabelSelector: labels.Set(job.ObjectMeta.Labels).String(),
+	}
+	// Stream Pod logs in the background
+	logsStreamingComplete := make(chan bool)
+	err = k.streamPodLogs(selector, op.Out, logsStreamingComplete)
+	if err != nil {
+		return err
+	}
+	// Watch job events and exit on failure/success
+	watch, err := k.jobs.Watch(selector)
+	if err != nil {
+		return err
+	}
+	for event := range watch.ResultChan() {
+		job, ok := event.Object.(*batchv1.Job)
+		if !ok {
+			return fmt.Errorf("unexpected type")
+		}
+		complete := false
+		for _, cond := range job.Status.Conditions {
+			if cond.Type == batchv1.JobFailed {
+				err = fmt.Errorf(cond.Message)
+				complete = true
+				break
+			}
+			if cond.Type == batchv1.JobComplete {
+				complete = true
+				break
+			}
+		}
+		if complete {
+			break
+		}
+	}
+
+	// Wait for pod logs to finish printing
+	for i := 0; i < int(k.requiredCompletions); i++ {
+		<-logsStreamingComplete
+	}
+
+	return err
+}
+
+// Handles receives an ImageType* and answers whether this driver supports that type
+func (k *KubernetesDriver) Handles(imagetype string) bool {
+	return imagetype == driver.ImageTypeDocker || imagetype == driver.ImageTypeOCI
+}
+
+func (k *KubernetesDriver) streamPodLogs(options metav1.ListOptions, out io.Writer, done chan bool) error {
+	watch, err := k.pods.Watch(options)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		watching := map[string]bool{}
+		for event := range watch.ResultChan() {
+			pod, ok := event.Object.(*v1.Pod)
+			if !ok {
+				continue
+			}
+			podName := pod.GetName()
+			if watching[podName] {
+				continue
+			}
+			req := k.pods.GetLogs(podName, &v1.PodLogOptions{
+				Container: k8sContainerName,
+				Follow:    true,
+			})
+			reader, err := req.Stream()
+			if err != nil {
+				continue
+			}
+
+			watching[podName] = true
+			io.Copy(out, reader)
+			reader.Close()
+			done <- true
+		}
+	}()
+
+	return nil
+}
+
+func (k *KubernetesDriver) deleteSecret(name string) error {
+	return k.secrets.Delete(name, &metav1.DeleteOptions{
+		PropagationPolicy: &k.deletionPolicy,
+	})
+}
+
+func (k *KubernetesDriver) deleteJob(name string) error {
+	return k.jobs.Delete(name, &metav1.DeleteOptions{
+		PropagationPolicy: &k.deletionPolicy,
+	})
+}
+
+// TODO: limit character count
```

Comment:
Was this meant to be done in this PR?

Reply:
I just double checked, and it looks like this gets automatically truncated by the API server.